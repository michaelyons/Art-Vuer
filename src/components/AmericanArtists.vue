<template>
  <div class="hello">
    <h1>{{msg}}</h1>
    <article v-for="artist in americanArtists" :key="artist.id">
      <h2>{{artist.name}}</h2>
      <p>{{artist.lifetime}}</p>
      <p>{{artist.artwork}}</p>
      <button :id="artist.id" v-on:click="select($event)">View Artwork</button>
    </article>
  </div>
</template>

<script>
import GetAmericanArtistsApi from '@/services/api/AmericanArtists';
import GetSpecificAmericanArtist from '@/services/api/SpecificAmerican';
export default {
  name: 'AmericanArtists',
  methods: {
    select: async event => {
      const targetId = event.target.id;
      const data = await GetSpecificAmericanArtist.getAmericanArtist(targetId);
      return data.records;
    }
  },
  data() {
    return {
      americanArtists: []
    };
  },
  created() {
    GetAmericanArtistsApi.getAmericanArtists()
      .then(artists => {
        this.americanArtists = artists;
      })
      .catch(error => console.log(error));
  },
  mounted() {
    GetAmericanArtistsApi.getAmericanArtists()
      .then(artists => {
        const artist = artists.map(artist => {
          return {
            name: artist.displayname,
            lifetime: artist.displaydate,
            artwork: artist.objectcount,
            website: artist.url,
            id: artist.id
          };
        });
        this.americanArtists = artist;
      })
      .catch(error => console.log(error));
  },
  props: {
    msg: String
  }
};
</script>

<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
article {
  border-style: solid;
  border-color: crimson;
  border-width: 5px;
  margin-bottom: 1rem;
}
a {
  color: #42b983;
}
</style>
