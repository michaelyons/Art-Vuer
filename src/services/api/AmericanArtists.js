import axios from "axios";

export default {
  getAmericanArtists() {
    return axios
      .get(
        `/person?q=culture:American&size=100&apikey=${process.env.VUE_APP_KEY}`
      )
      .then(response => {
        return response.data.records;
      });
  }
};
