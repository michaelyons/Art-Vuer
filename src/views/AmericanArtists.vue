<template>
  <div class="artists">
    <AmericanArtists msg="American Artists"/>
  </div>
</template>

<script>
import AmericanArtists from '@/components/AmericanArtists.vue';
export default {
  name: 'american-artists',
  components: {
    AmericanArtists
  }
};
</script>
