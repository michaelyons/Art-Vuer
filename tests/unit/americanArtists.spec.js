import { shallowMount } from "@vue/test-utils";
import AmericanArtists from "@/components/AmericanArtists.vue";

describe("AmericanArtists.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(AmericanArtists);
    expect(wrapper).toMatchSnapshot();
  });
});
