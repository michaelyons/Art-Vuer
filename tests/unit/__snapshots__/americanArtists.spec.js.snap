// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmericanArtists.vue matches snapshot 1`] = `
<div>
  <div class="hello">
    <div class="art"></div>
    <div class="pictures"></div>
  </div>
</div>
`;